libmariadb/mariadb_stmt.c

```diff
@@ -790,9 +790,17 @@ unsigned char* mysql_stmt_execute_generate_simple_request(MYSQL_STMT *stmt, size
         case MYSQL_TYPE_ENUM:
         case MYSQL_TYPE_BIT:
         case MYSQL_TYPE_SET:
-          size+= 5; /* max 8 bytes for size */
+          size+= 9; /* max 9 bytes for size */
           size+= (size_t)ma_get_length(stmt, i, 0);
           break;
+        case MYSQL_TYPE_DATE:
```

Comment:
These types represent a binary type (MYSQL_TIME), so why do you set the length to the corresponding string representation?

Reply:
Maybe the used constants are not correct, but those are the constants store_param uses to allocate memory.
I had to handle the Time cases, because otherwise the default case adds "-1" to the size, whitch can lead to a heap corruption in some cases, when space is reallocated.